### PR TITLE
fix(image_projection_based_fusion): fix unexpected matches between roi and cluster

### PR DESCRIPTION
**Pull request by yukke42**
## Description

- fix a bug of unexpected matches between roi and cluster by the default index is 0

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [ ] I've confirmed the [contribution guidelines].
- [ ] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].
- [ ] The PR has been properly tested.
- [ ] The PR has been reviewed by the code owners.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.
- [ ] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### Summary by CodeRabbit

Bug fix: Fixed a bug in `roi_cluster_fusion` where unexpected matches occurred between regions of interest (ROI) and clusters. The bug was caused by an incorrect default index, which has been changed to -1. Additionally, a check has been added to skip processing if the index remains -1.

> "Bugs squashed, ROI and clusters aligned,
> No more surprises, peace of mind.
> Default index fixed, -1 it shall be,
> Skip processing when it's all we see."

### CODE REVIEW

**Comment by yukke42** [comment]
https://github.com/autowarefoundation/autoware.universe/pull/4444

**Review by miursh — APPROVED** [review]
LGTM

File: perception/image_projection_based_fusion/src/roi_cluster_fusion/node.cpp [hunk 1]
```diff
@@ -169,7 +169,7 @@ void RoiClusterFusionNode::fuseOnSingleImage(
   }
 
   for (const auto & feature_obj : input_roi_msg.feature_objects) {
-    int index = 0;
+    int index = -1;
```
Comment (yukkysaito) [inline perception/image_projection_based_fusion/src/roi_cluster_fusion/node.cpp:172]: How about the following?

```suggestion

    int index = 0;
    bool associated = false;
    double max_iou = 0.0;
    for (const auto & cluster_map : m_cluster_roi) {
      double iou(0.0), iou_x(0.0), iou_y(0.0);
@ -191,6 +191,10 @@ void RoiClusterFusionNode::fuseOnSingleImage(
        max_iou = iou + iou_x + iou_y;
      }
    }
    if (!associated) {
      continue;
    }

```

Comment (yukke42) [reply perception/image_projection_based_fusion/src/roi_cluster_fusion/node.cpp:172]: @yukkysaito Thanks for suggestion. It applied in [e83de7a](https://github.com/autowarefoundation/autoware.universe/pull/4445/commits/e83de7aff03217cc347fff296c1523580d27d21a).

**Comment by yukke42** [comment]
I will fix conflict soon.

**Pull request merged**